### DISCUSSION TITLE
Oneway builder

### DISCUSSION THREAD

**Discussion by michaldo** [Randgalt/record-builder#140] [Ideas]
The benefit of builders is readability, but there is a risk of missing parameter. Constructor is less readable, but compiler ensures all parameters are present.

Did you consider oneway builder?

Example:
`public record NameAndAge(String name, int age){}`

Oneway builder is something like
```
interface NameAndAgeBuilder {
  AgeBuilder name(String name);
}
interface AgeBuilder {
  NameAndAge age(int age);
}

usage:
NameAndAge foo123 = NameAndAgeBuilder.builder().name("foo").age(123);
```

Oneway builder has both: readability and completeness. Completeness, because to get `NameAndAge` you must follow path `name(...).age(...)` and there is no other path to create NameAndAge object.

**Comment by Randgalt** [comment]
That would be a lot of extra classes to make work. I think it's out of the scope of this project.
